ipn/ipnlocal/serve.go

```diff
@@ -700,40 +624,42 @@ func (b *LocalBackend) addTailscaleIdentityHeaders(r *httputil.ProxyRequest) {
 	r.Out.Header.Set("Tailscale-Headers-Info", "https://tailscale.com/s/serve-headers")
 }
 
-func (b *LocalBackend) serveWebHandler(w http.ResponseWriter, r *http.Request) {
-	h, mountPoint, ok := b.getServeHandler(r)
-	if !ok {
-		http.NotFound(w, r)
-		return
-	}
-	if c, ok := getServeHTTPContext(r); ok {
-		b.maybeLogServeConnection(c.DestPort, c.SrcAddr)
-	}
-	if s := h.Text(); s != "" {
-		w.Header().Set("Content-Type", "text/plain; charset=utf-8")
-		io.WriteString(w, s)
-		return
-	}
-	if v := h.Path(); v != "" {
-		b.serveFileOrDirectory(w, r, v, mountPoint)
-		return
-	}
-	if v := h.Proxy(); v != "" {
-		p, ok := b.serveProxyHandlers.Load(v)
+func (b *LocalBackend) serveWebHandler(sessionID string) http.HandlerFunc {
```

Comment:
docs + rename to newServeWebHandlerForSession? and document that sessionID can be empty and what tha means
```suggestion
func (b *LocalBackend) serveWebHandler(sessionID string) http.HandlerFunc {
```
```suggestion
func (b *LocalBackend) serveWebHandler(sessionID string) http.HandlerFunc {
```